python/lsst/sims/catUtils/mixins/snObject.py

```diff
@@ -0,0 +1,718 @@
+        Returns
+        -------
+        float
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        SEDfromSNcosmo = self.SNObjectSED(time=time,
+                                          bandpass=bandpassobject)
+        return SEDfromSNcosmo.calcFlux(bandpass=bandpassobject) / 3631.0
+
+    def catsimBandMagError(self, time, bandpassobject, m5, photParams=None,
+                           magnitude=None):
+        """
+        return the 68 percent uncertainty on the magnitude in the bandpass
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassobject: mandatory, `lsst.sims.photUtils.BandPass` object
+            A particular bandpass which is an instantiation of one of
+            (u, g, r, i, z, y)
+        m5 :
+        photParams :
+        magnitude :
+        Returns
+        -------
+        float
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+
+        if magnitude is None:
+            mag = self.catsimBandMags(time=time, bandpassobject=bandpassobject)
+        else:
+            mag = magnitude
+
+        # mag = np.asarray([[mag]])
+        bandpass = bandpassobject
+
+        if photParams is None:
+            photParams = PhotometricParameters()
+
+        # m5 = np.asarray([[m5]])
+
+        magerr = calcMagError_m5(magnitude=mag, bandpass=bandpass, m5=m5,
+                                 photParams=photParams)
+        return magerr[0]
+
+    def catsimBandFluxError(self, time, bandpassobject, m5, photParams=None,
+                            magnitude=None):
+        """
+        return the flux uncertainty in the bandpass in units 'maggies'
+        (the flux the AB magnitude reference spectrum would have in the
+        same band.)
+
+        Parameters
+        ----------
+        time: mandatory, float
+            MJD at which band fluxes are evaluated
+        bandpassobject: mandatory, `lsst.sims.photUtils.BandPass` object
+            A particular bandpass which is an instantiation of one of
+            (u, g, r, i, z, y)
+        m5 :
+        photParams :
+        magnitude :
+        Returns
+        -------
+        float
+
+        Examples
+        --------
+        .. note: If there is an unphysical value of sed in
+        the wavelength range, it produces a flux of  `np.nan`
+        """
+        # SEDfromSNcosmo = self.SNObjectSED(time=time,
+        #                                 bandpass=bandpassobject)
+
+        if magnitude is None:
+            mag = self.catsimBandMags(time=time, bandpassobject=bandpassobject)
+        else:
+            mag = magnitude
+        flux = self.catsimBandFluxes(time=time, bandpassobject=bandpassobject)
```

Comment:
calling catsimBandMags and catsimBandFluxes means you are integrating the SED over the Bandpass twice.  Can you just do one and use the known relationship between flux and magnitude to get the other?